File: medium/subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<int> subset;
    void gen(int n, vector<int> nums) {
        if (n==size(nums)) {
            res.push_back(subset);
            return;
        }
        subset.push_back(nums[n]);
        gen(n+1,nums);
        subset.pop_back();
        gen(n+1,nums);
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        gen(0,nums);
        return res;
    }
};

File: medium/subsets/subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        subset=[]
        def gen(i):
            if i==len(nums):
                res.append(subset.copy())
                return
            subset.append(nums[i])
            gen(i+1)
            subset.pop()
            gen(i+1)
        gen(0)
        return res
